mods/chat.js

```diff
@@ -1,573 +1,603 @@
-            labels[counter].setAttribute ( 'class', cur.replace ( /(^|\s)active(\s|$)/, ' ' ) );
-            tabs[counter].setAttribute ( 'class', '' );
-            if ( tabs[counter].getAttribute ( 'data-id' ) === element.getAttribute ( 'data-id' ) ) {
-                tabs[counter].setAttribute ( 'class', 'active' );
+            var cur = labels[counter].getAttribute('class') + '';
+            labels[counter].setAttribute('class', cur.replace(/(^|\s)active(\s|$)/, ' '));
+            tabs[counter].setAttribute('class', '');
+            if (tabs[counter].getAttribute('data-id') === element.getAttribute('data-id')) {
+                tabs[counter].setAttribute('class', 'active');
             }
         }
-        if ( element.hasAttribute ( 'class' ) ) {
-            element.setAttribute ( 'class', ( element.getAttribute ( 'class' ) ).replace ( /(^|\s)new-message(\s|$)/, ' ' ) + ' active' );
+        if (element.hasAttribute('class')) {
+            element.setAttribute('class', (element.getAttribute('class')).replace(/(^|\s)new-message(\s|$)/, ' ') + ' active');
         } else {
-            element.setAttribute ( 'class', 'active' );
+            element.setAttribute('class', 'active');
         }
     },
-    openCloseChat: function ( element ) {
+    openCloseChat: function (element) {
         var chat = element.parentNode;
-        if ( chat.getAttribute ( 'class' ) === 'idrinth-hovering-box active' || chat.getAttribute ( 'class' ) === 'idrinth-hovering-box active left-sided' ) {
-            chat.setAttribute ( 'class', 'idrinth-hovering-box' + ( idrinth.settings.moveLeft ? ' left-sided' : '' ) +
-                    ( chat.getElementsByClassName ( 'new-message' ) && chat.getElementsByClassName ( 'new-message' ).length ? ' new-message' : '' ) );
-            element.innerHTML = '&lt;&lt;';
+        if (chat.getAttribute('class') === 'idrinth-hovering-box active' || chat.getAttribute('class') === 'idrinth-hovering-box active left-sided') {
+            chat.setAttribute('class', 'idrinth-hovering-box' + ( idrinth.settings.moveLeft ? ' left-sided' : '' ) +
+                ( chat.getElementsByClassName('new-message') && chat.getElementsByClassName('new-message').length ? ' new-message' : '' ));
+        } else {
+            chat.setAttribute('class', 'idrinth-hovering-box active' + ( idrinth.settings.moveLeft ? ' left-sided' : '' ));
+        }
+        element.innerHTML = '&gt;&gt;';
+    },
+    getMsg: function(key){
+        var textKey = key || '';
+        var text = {
+            'modify.fail' : 'Can\'t modify that user at the moment',
+            'create.fail' : 'Can\'t create at the moment',
+            'join.fail': 'Can\'t join at the moment',
+            'join.notwork':'Joining didn\'t work',
+            'user.unknown':'The given username for dotd.idrinth.de is unknown, do you want to register it there?',
+            'login.fail':'Login failed in an unexpected way'
+        };
+        return text[textKey];

```

Comment:
lacks handling of unknown key and the whole message lookup might be better placed somewhere central, allowing for easierr reuse